controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -171,11 +172,13 @@ func (r *StorageClassRequestReconciler) SetupWithManager(mgr ctrl.Manager) error
 			return []reconcile.Request{}
 		})
 	enqueueForOwner := handler.EnqueueRequestForOwner(mgr.GetScheme(), mgr.GetRESTMapper(), &v1alpha1.StorageClassRequest{})
+	generationChangedPredicate := builder.WithPredicates(predicate.GenerationChangedPredicate{})
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.StorageClassRequest{}, builder.WithPredicates(
 			predicate.GenerationChangedPredicate{},
 		)).
-		Watches(&rookCephv1.CephBlockPool{}, enqueueForOwner).
+		Owns(&rookCephv1.CephBlockPoolRadosNamespace{}, generationChangedPredicate).
 		Watches(&rookCephv1.CephFilesystemSubVolumeGroup{}, enqueueForOwner).
```

Comment:
1. Should we remove the watch over cephBlockPool? Can a blockpool be deleted if rados namespace exists? 
2. Why are we Owning RadosNamespace and Watching SubVolumeGroup? shouldn't we either Own both of them or Watch both of them?

Reply:
1. We should, so I did. And no, Rook-Ceph keeps a finalizer on the CephBlockPool CR until (among other conditions) there are no more RADOS namespaces defined for the pool.
2. See this (resolved) comment: https://github.com/red-hat-storage/ocs-operator/pull/2478#discussion_r1521870114 Has to do with the recent-ish change to using OwnerRefs.